cfme/tests/cloud/test_provisioning.py

```diff
@@ -50,24 +42,59 @@ def test_provision_from_template(request, setup_provider, provider, vm_name, pro
         'first_name': 'Image',
         'last_name': 'Provisioner',
         'notes': note,
-        'instance_type': provisioning['instance_type'],
-        'availability_zone': provisioning['availability_zone'],
-        'security_groups': [provisioning['security_group']],
-        'guest_keypair': provisioning['guest_keypair']
+        'instance_type': provisioning['instance_type'] if provider.type != "azure" else
+        provisioning['vm_size'],
+        "availability_zone": provisioning["availability_zone"] if provider.type != "azure" else
+        None,
+        'security_groups': [provisioning['security_group']] if provider.type != "azure" else
+        provisioning["network_nsg"],
+        "guest_keypair": provisioning["guest_keypair"] if provider.type != "azure" else None
     }
 
     if isinstance(provider, OpenStackProvider):
         inst_args['cloud_network'] = provisioning['cloud_network']
 
     if isinstance(provider, AzureProvider):
-        inst_args['environment__cloud_network'] = provisioning['cloud_network']
-        inst_args['environment__cloud_subnet'] = provisioning['cloud_subnet']
-        inst_args['environment__security_groups'] = provisioning['security_group']
-        inst_args['environment__resource_group'] = provisioning['resource_group']
-        inst_args['hardware__instance_type'] = provisioning['vm_size']
+        inst_args['cloud_network'] = provisioning['virtual_net']
+        inst_args['cloud_subnet'] = provisioning['subnet_range']
+        inst_args['security_groups'] = provisioning['network_nsg']
+        inst_args['resource_groups'] = provisioning['resource_group']
+        inst_args['instance_type'] = provisioning['vm_size']
+        inst_args['admin_username'] = provisioning['vm_user']
+        inst_args['admin_password'] = provisioning['vm_password']
+    return instance, inst_args
+
 
+@pytest.fixture(scope="function")
+def vm_name(request):
+    vm_name = 'test_image_prov_{}'.format(fauxfactory.gen_alphanumeric())
+    return vm_name
+
+
+@pytest.mark.tier(2)
+def test_provision_from_template(request, setup_provider, provider, testing_instance, soft_assert):
+    """ Tests instance provision from template
+
+    Metadata:
+        test_flag: provision
+    """
+    instance, inst_args = testing_instance
     sel.force_navigate("clouds_instances_by_provider")
     instance.create(**inst_args)
+    if instance.provider.type == "azure":
```

Comment:
Let's just leave this at 800 for all, it's preferable to not have to deal with provider differences in tests unless we have to